### DISCUSSION TITLE
Fatal Error with new docker install - v5

### DISCUSSION THREAD

**Discussion by 8bitgentleman** [firefly-iii/firefly-iii#8133] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Hello, I have just booted up a new docker-compose of firefly iii using the instruction in the [docs here](https://docs.firefly-iii.org/firefly-iii/installation/docker/). After creating a login-password when attempting to create my first asset account firefly crashes with a fatal PHP error. See below for the stack trace. I've run firefly iii many times before but now with new installs I just keep getting this error.

I am on v5 of firefly because I am on an older Synology NAS which it seems is not supported by firefly v6?
If you have any idea what's going on here it would be much appreciated

```
Truncated from this point <----|[2023-11-05 20:26:53] local.ERROR: Expected response code "250" but got code "530", with message "530 5.7.1 Authentication required".  
[2023-11-05 20:27:23] local.ERROR: Exception is: {"class":"Symfony\\Component\\ErrorHandler\\Error\\FatalError","errorMessage":"Declaration of FireflyIII\\Support\\Request\\ConvertsDataTypes::string(string $field): string must be compatible with Illuminate\\Http\\Request::string($key, $default = null)","time":"Sun, 05 Nov 2023 20:27:23 -0500","file":"\/var\/www\/html\/app\/Http\/Requests\/NewUserFormRequest.php","line":54,"code":0,"version":"5.7.3","url":"https:\/\/firefly.treasurechest.design\/new-user\/submit","userAgent":"Mozilla\/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/119.0.0.0 Safari\/537.36","json":true,"method":"POST"}  
[2023-11-05 20:27:23] local.ERROR: Exception when mailing: Expected response code "250" but got code "530", with message "530 5.7.1 Authentication required".  
[2023-11-05 20:27:23] local.ERROR: Declaration of FireflyIII\Support\Request\ConvertsDataTypes::string(string $field): string must be compatible with Illuminate\Http\Request::string($key, $default = null) {"userId":1,"exception":"[object] (Symfony\\Component\\ErrorHandler\\Error\\FatalError(code: 0): Declaration of FireflyIII\\Support\\Request\\ConvertsDataTypes::string(string $field): string must be compatible with Illuminate\\Http\\Request::string($key, $default = null) at /var/www/html/app/Http/Requests/NewUserFormRequest.php:54)
[stacktrace]
#0 {main}
"} 

```

### Debug information

Debug information generated at 2023-11-05 20:27:42 America/New_York for Firefly III version **5.7.3**.

| Scope | Version |
| --- | --- |
| Firefly III | 5.7.3 |
| Firefly III API | 1.5.6 |
| Build | 168, 02-05-2022 19:37:51 CEST |
| DB version | 18 (exp. 18) |
| PHP | `8.0.17` |
| Host | `Linux` |

| System info | Value |
| --- | --- |
| System TZ | America/New_York |
| Browser TZ | America/New_York |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | notice, daily |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | apache2handler |
| Default language | en_US |
| Default locale | equal |
| BCscale | 24 |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | mysql |
| Trusted proxies (.env) | `**` |

| User info | Value |
| --- | --- |
| Login provider | eloquent  |
| User guard | web  |
| Headers | N/A, N/A  |
| Stateful domains |  |
| Session start | 2023-11-01 00:00:00 |
| Session end | 2023-11-30 23:59:59 |
| Session first | 2023-01-01 00:00:00 |
| User ID | 1 |
| User language | en_US |
| User locale | en_US |
| Attempt at "en_US.utf8" | 'en_US.utf8' |
| Attempt at "en_US.UTF-8" | 'en_US.UTF-8' |
| User agent | Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36 |

### Expected behaviour

I should be able to create accounts

### Steps to reproduce

Start up a new docker container. Follow the new account instructions

**Comment by JC5** [comment]
Sorry, I also don’t know why your synology box did not pull the latest image.

**Reply by 8bitgentleman** [reply]
Hi James I'm running docker on a Synology NAS that has an older kernel it seems. I've run v5 successfully for a long time in the past. When trying to upgrade to v6 I run into the 'random number generator' error described in #7115 . That says to me I can't use v6 correct? Now trying to revert and use my old docker-compose for v5 I'm running into the error above. 

I understand v5 is an old version but that seems to be the only way forward since v5 worked smoothly before? Is there anyway for me to go back to v5? I'm fine with starting fresh

**Reply by JC5** [reply]
You're on 5.x right now, correct? You can start fresh on 5.x but if it doesn't work out of the box there's not much I can do for you I'm afraid.

**Reply by 8bitgentleman** [reply]
I understand, there's only so much support you can give to an old version. What's confusing to me is the sequence of events

1. 5.x works fine
2. I attempted to upgrade to 6.x and encountered the error described in #7115 
3. I removed the 6.x container and went back to my 5.x docker-compose file
4. 5.x now doesn't work

Does the 6.x image do something that wouldn't be undone/removed with a simple docker-compose rm and docker prune? I'm thinking there may be leftovers of the 6.x install somewhere on my system that are causing issues

**Reply by JC5** [reply]
It should leave nothing behind, but they are both based on "fireflyiii/base" and both images point to "latest". That could be it, I'm not sure if I can fix that.

**Reply by 8bitgentleman** [reply]
I'm targeting a 5.x in the docker-compose like this

`image: jc5x/firefly-iii:5.7.3`
Is there some way to do that with fireflyiii/base?  Is there versioning with fireflyiii/base

**Reply by JC5** [reply]
There is not :(

**Reply by 8bitgentleman** [reply]
For my own understanding where is fireflyiii/base used and if it isn't versioned what is it for?

**Reply by JC5** [reply]
It is versioned, I had to check. But it's versioned on the minor PHP version, not on the Firefly III version. And it has been php 8.2 for a while now.

**Reply by JC5** [reply]
Wait that makes no sense, the base image installs nothing. It's just php. The errors seem to come from somewhere else. I have no idea.
